content/docs/gke/private-clusters.md

```diff
@@ -7,60 +7,176 @@ weight = 5
+    kfctl generate all -V
+    ```
+   * **${KFAPP}** - the _name_ of a directory where you want Kubeflow 
+     configurations to be stored. This directory is created when you run
+     `kfctl init`. If you want a custom deployment name, specify that name here.
+     The value of this variable becomes the name of your deployment.
+     The value of this variable cannot be greater than 25 characters. It must
+     contain just the directory name, not the full path to the directory.
+     The content of this directory is described in the next section.
+   * **${PROJECT}** - the _name_ of the GCP project where you want Kubeflow 
+     deployed.
+   * When you run `kfctl init` you need to choose to use either IAP or basic 
+     authentication, as described below.
+   * `kfctl generate all` attempts to fetch your email address from your 
+     credential. If it can't find a valid email address, you need to pass a
+     valid email address with flag `--email <your email address>`. This email 
+     address becomes an administrator in the configuration of your Kubeflow 
+     deployment.
+
+1. Enable private clusters by editing `${KFAPP}/gcp_configs/cluster-kubeflow.yaml` and updating the following two parameters:
 
     ```
     privatecluster: true
     gkeApiVersion: v1beta1
     ```
+1. Remove components which are not useful in private clusters:
+
+    ```
+    cd ${KFAPP}/ks_app
+    ks component rm cert-manager
+    ```
 1. Create the deployment:
 
     ```
     cd ${KFAPP}
     kfctl apply platform
     ```
 
-1. To set up ingress to the cluster, it is recommended to use your custom domain instead of Cloud Endpoints. cert-manager cannot be used to create HTTPS certificates because cert-manager needs to talk to LetsEncrypt to get the certificate and that is not possible in a private cluster setting. Obtain the HTTPS certificates for your ${FQDN} and create a k8s secret with it. Assuming your cert and key are present in files named tls.crt and tls.key, create a secret using the following command:
+   * If you get an error **legacy networks not supported** Follow the 
+     [troubleshooting guide]( /docs/gke/troubleshooting-gke/#legacy-networks-are-not-supported) to create a new network.
 
-    ```
-    kubectl create secret generic --namespace=${NAMESPACE} envoy-ingress-tls --from-file=tls.crt=tls.crt --from-file=tls.key=tls.key
-    ```
+     * You will need to manually create the network as a work around for [kubeflow/kubeflow#3071](https://github.com/kubeflow/kubeflow/issues/3071)
+
+       ```
+       cd ${KFAPP}/gcp_configs
+       gcloud --project=${PROJECT} deployment-manager deployments create ${KFAPP}-network --config=network.yaml
+       ```
+
+     * Then edit **gcp_config/cluster.jinja** to add a field **network** in your cluster
+     
+       ```
+       cluster:
+          name: {{ CLUSTER_NAME }}
+          network: <name of the new network>
+       ```
+   
+     * To get the name of the new network run
+      
+       ```
```

Comment:
Same indentation issue with this code block - needs an extra level of indentation in the source.